src/Forms/GridField/GridFieldDetailForm.php

```diff
@@ -106,15 +107,33 @@ public function __construct($name = null, $showPagination = null, $showAdd = nul
      */
     public function handleItem($gridField, $request)
     {
-        if ($gridStateStr = $request->getVar('gridState')) {
-            $gridField->getState(false)->setValue($gridStateStr);
-        }
-
         // Our getController could either give us a true Controller, if this is the top-level GridField.
         // It could also give us a RequestHandler in the form of GridFieldDetailForm_ItemRequest if this is a
         // nested GridField.
         $requestHandler = $gridField->getForm()->getController();
+        $record = $this->getRecordFromRequest($gridField, $request);
```

Comment:
Interesting. This method is just a simple encapsulation of the logic that was already there. As it is now, if the record comes back null (`->byID($nonExistentID)`), it blows up when the RequestHandler tries to access properties on `null`.

We need to flag this as a separate issue. Rare, but possible, if someone has a saved link to a record that has since been deleted.

Reply:
Actually, never mind. I've added a simple sanity check and `404`.